app/src/androidTest/java/com/droidhats/campuscompass/views/MapFragmentTest.kt

```diff
@@ -54,12 +64,211 @@ class MapFragmentTest {
         }
     }
 
-    //Checking if the search bar exists
+
     @Test
     fun test_SearchBar() {
+        //Checks if the search bar with text "search" exists
+        val searchTextView = onView(
+            allOf(
+                withId(R.id.mt_placeholder), withText("Search"), isDisplayed()
+            )
+        )
+        searchTextView.check(matches(withText("Search")))
+
+        //Checks if magnifying glass button is displayed
+        val magnifyingGlassView = onView(
+            allOf(
+                withId(R.id.mt_search), isDisplayed()
+            )
+        )
+        magnifyingGlassView.check(matches(isDisplayed()))
+
+        //Checks if Search bar can be clicked
+        val containerSearchBar = onView(
+            allOf(
+                withId(R.id.mapFragSearchBar), isDisplayed()
+            )
+        )
+        containerSearchBar.perform(click())
```

Comment:
As mentioned earlier, avoid declaring variable for onView method specially since it is not being reused.